### PR TITLE
Replacement of XML tests

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
This PR is an intermediate step to help us with #1637

This was done by: adding a method to apply a function over all testcases, and by adding a `save_test()` into BaseTest (now named TestCase) that overwrites any tests in our current spec. So it was basically invoking apply_transform_over_tests with a lambda that invokes `save_test()`

In Summary:

1. All tests are now unified in a TestCase class
2. Removed the capacity of running all unit tests directly from testing_tool (this functionality was deprecated in favour of using CMake to find the tests)
3. Added a function to apply a function over all test cases.

### CODE REVIEW

File: regression/testing_tool.py [hunk 1]
```diff
@@ -381,3 +287,29 @@ def main():
...
     main()
+
+# Utilities for REPL. This helps us to do batch changes over all tests.
+# Example:
+# 1. Invoke python3 in shell
+# 2. `import testing_tool`
+# 3. `testing_tool.apply_transform_over_tests(testing_tool.print_test)`
+
+# TODO We probably should obtain this from CMake somehow
+TEST_SUITES = [ "bitwuzla", "cvc", "mathsat", "smtlib", "z3", "esbmc-solidity", "Interval-analysis-ibex-contractor",
```
Comment (fbrausse) [inline regression/testing_tool.py]: Re comment: yes, we should. People create new sub-directories all the time.

Comment (rafaelsamenezes) [reply regression/testing_tool.py]: I guess we could change the transform function to work with a directory and use CMake to invoke a wrapper of the transform.

Comment (fbrausse) [reply regression/testing_tool.py]: Sounds complicated. As this is for manual use only, I'm fine leaving this list hardcoded for now. Maybe add a comment to check whether the list is still up to date for manual invocations of the new function.

File: regression/testing_tool.py [hunk 2]
```diff
@@ -381,3 +287,29 @@ def main():
...
+                "esbmc-old", "jimple", "python", "cheri-c", "cheri-128", "esbmc-cpp/cpp", "esbmc-cpp/cbmc", "esbmc-cpp/polymorphism_bringup",
+                "esbmc-cpp/inheritance_bringup", "esbmc-cpp/bug_fixes", "esbmc-cpp/OM_sanity_checks", "esbmc-cpp/string", "esbmc-cpp/stream",
+                "esbmc-cpp/gcc-template-tests", "esbmc-cpp/template", "esbmc-cpp/algorithm", "esbmc-cpp/deque", "esbmc-cpp/list", "esbmc-cpp/map",
+                "esbmc-cpp/multimap", "esbmc-cpp/multiset", "esbmc-cpp/priority_queue", "esbmc-cpp/queue", "esbmc-cpp/set", "esbmc-cpp/stack",
+                "esbmc-cpp/unix", "esbmc-cpp/vector", "esbmc-cpp11/cpp", "esbmc-cpp11/constructors", "esbmc-cpp11/new-delete", "esbmc-cpp11/reference",
+                "esbmc-unix", "esbmc-unix2", "esbmc", "cbmc", "cstd", "llvm", "floats", "floats-regression", "k-induction", "csmith", "k-induction-parallel",
+                "cuda/benchmarks", "cuda/Supported_long_time", "nonz3", "incremental-smt", "extensions", "cuda/COM_sanity_checks", "linux"]
```
Comment (fbrausse) [inline regression/testing_tool.py]: It's easier to work with git or other line-based tools when such lists (a) have just a single entry per line and, less importantly, (b) are ordered (alphabetically, for instance). Fewer merge conflicts.

**Review by fbrausse — APPROVED** [review]
Thanks Rafael, this is a good initiative.

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @rafaelsamenezes.

**Pull request merged**